### ISSUE TITLE
Treinar YOLOv8 com dados anotados

### ISSUE THREAD

**Issue by afmyhouse** [SEAME-pt/T07-ADS-Object_Detection_and_Avoidance#8]

**Comment by afmyhouse** [comment]
<html><head></head><body><p><strong>"Training YOLOv8 with annotated data"</strong> broken down in <strong>Markdown format</strong>, with every step clarified from setup to training to monitoring.</p>
<hr>
<h2>🏋️ Task: Training YOLOv8 with Annotated Data</h2>
<h3>🎯 Goal</h3>
<p>Train a YOLOv8 model using your <strong>custom annotated dataset</strong> (exported from Roboflow, CVAT, or manually). The model will learn to detect specific objects relevant to your project (e.g., lanes, obstacles, road signs).</p>
<hr>
<h2>⚙️ Requirements</h2>

Component | Description
-- | --
Python 3.8+ | Core requirement
PyTorch | Needed for Ultralytics YOLO backend
Ultralytics lib | Install via pip install ultralytics
Annotated Data | In YOLO format with images/, labels/, and data.yaml


<hr>
<h2>🗂️ Dataset Structure</h2>
<p>Make sure your exported dataset looks like this:</p>
<pre><code>datasets/custom_yolo/
├── images/
│   ├── train/
│   └── val/
├── labels/
│   ├── train/
│   └── val/
└── data.yaml
</code></pre>
<h3>Example <code inline="">data.yaml</code></h3>
<pre><code class="language-yaml">path: datasets/custom_yolo
train: images/train
val: images/val
names: ['car', 'pedestrian', 'traffic_light']
</code></pre>
<hr>
<h2>🚀 Training Script</h2>
<p>Create a script named <code inline="">train_custom_yolo.py</code>:</p>
<pre><code class="language-python">from ultralytics import YOLO

# Load model (n = nano, s = small, m = medium, l = large)
model = YOLO("yolov8n.pt")  # Start from a pre-trained checkpoint

# Train
model.train(
    data="datasets/custom_yolo/data.yaml",
    epochs=50,
    imgsz=640,
    batch=16,
    project="runs/train",
    name="yolov8_custom",
    resume=False  # Set to True if continuing training
)
</code></pre>
<hr>
<h2>📈 Output</h2>
<p>After training finishes:</p>
<pre><code class="language-bash">runs/train/yolov8_custom/
├── weights/
│   ├── best.pt     # Best model (use for inference)
│   └── last.pt     # Last epoch checkpoint
├── results.png     # Training/validation loss curves
└── confusion_matrix.png
</code></pre>
<hr>
<h2>✅ Task Checklist</h2>
<ul class="contains-task-list">
<li class="task-list-item">
<p><input type="checkbox" checked="" disabled=""> Dataset in YOLO format, validated</p>
</li>
<li class="task-list-item">
<p><input type="checkbox" checked="" disabled=""> <code inline="">data.yaml</code> correctly configured</p>
</li>
<li class="task-list-item">
<p><input type="checkbox" checked="" disabled=""> YOLOv8 training runs successfully</p>
</li>
<li class="task-list-item">
<p><input type="checkbox" checked="" disabled=""> <code inline="">best.pt</code> model exported</p>
</li>
<li class="task-list-item">
<p><input type="checkbox" checked="" disabled=""> Results analyzed (loss curves, precision, recall)</p>
</li>
<li class="task-list-item">
<p><input type="checkbox" checked="" disabled=""> (Optional) Evaluate model on KITTI, COCO frames</p>
</li>
</ul>
<hr>
<h2>🧠 Optional Enhancements</h2>
<ul>
<li>
<p>Add <strong>early stopping</strong> or custom <strong>learning rate scheduler</strong></p>
</li>
<li>
<p>Train on Jetson Nano (using TensorRT quantized models later)</p>
</li>
<li>
<p>Integrate <strong>WandB</strong> or <strong>TensorBoard</strong> for live monitoring</p>
</li>
<li>
<p>Export to <code inline="">ONNX</code> or <code inline="">TensorRT</code> after training for embedded use</p>
</li>
</ul>
<hr>
</body></html>
